src/main/java/mServer/crawler/sender/MediathekSrf.java

```diff
@@ -205,23 +206,44 @@ private void addFilms(String jsonMovieUrl, String themePageUrl, String theme) {
                 String url_small = "";
 
                 String url3u8 = urlHD.endsWith("m3u8") ? urlHD : url_normal;
+
                 if (url3u8.endsWith("m3u8")) {
                     m3u8Page = getUrl.getUri_Utf(SENDERNAME, url3u8, m3u8Page, "");
-                    if (m3u8Page.indexOf(INDEX_5) != -1) {
-                        urlHD = getUrlFromM3u8(url3u8, INDEX_5);
-                    }
-
-                    if (m3u8Page.indexOf(INDEX_4) != -1) {
-                        url_normal = getUrlFromM3u8(url3u8, INDEX_4);
-                    } else if (m3u8Page.indexOf(INDEX_3) != -1) {
-                        url_normal = getUrlFromM3u8(url3u8, INDEX_3);
+                    if (m3u8Page.length() == 0) {
+                        // tauschen https://srfvodhd-vh.akamaihd.net http://hdvodsrforigin-f.akamaihd.net
+                        // ist ein 403
+                        if (url3u8.startsWith("https://srfvodhd-vh.akamaihd.net")) {
+                            url3u8 = url3u8.replaceFirst("https://srfvodhd-vh.akamaihd.net", "http://hdvodsrforigin-f.akamaihd.net");
+                            m3u8Page = getUrl.getUri_Utf(SENDERNAME, url3u8, m3u8Page, "");
+                        }
                     }
-                    if (m3u8Page.indexOf(INDEX_2) != -1) {
-                        url_small = getUrlFromM3u8(url3u8, INDEX_2);
-                    } else if (m3u8Page.indexOf(INDEX_1) != -1) {
-                        url_small = getUrlFromM3u8(url3u8, INDEX_1);
+                    if (url3u8.contains("q50,q60")) {
+                        if (m3u8Page.indexOf(INDEX_5) != -1) {
+                            urlHD = getUrlFromM3u8(url3u8, INDEX_5);
+                        }
+                        if (m3u8Page.indexOf(INDEX_4) != -1) {
+                            url_normal = getUrlFromM3u8(url3u8, INDEX_4);
+                        } else if (m3u8Page.indexOf(INDEX_3) != -1) {
+                            url_normal = getUrlFromM3u8(url3u8, INDEX_3);
+                        }
+                        if (m3u8Page.indexOf(INDEX_2) != -1) {
+                            url_small = getUrlFromM3u8(url3u8, INDEX_2);
+                        } else if (m3u8Page.indexOf(INDEX_1) != -1) {
+                            url_small = getUrlFromM3u8(url3u8, INDEX_1);
+                        }
+                    } else {
+                        System.out.println(url3u8);
+                        if (m3u8Page.indexOf(INDEX_0) != -1) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)